src/raftstore/store/worker/pd.rs

```diff
@@ -273,12 +243,53 @@ impl<T: PdClient> Runner<T> {
         });
         handle.spawn(f);
     }
+
+    fn schedule_heartbeat_receiver(&mut self, handle: &Handle) {
+        let ch = self.ch.clone();
+        let f = self.pd_client
+            .handle_region_heartbeat_response(self.store_id, move |mut resp| {
+                PD_REQ_COUNTER_VEC.with_label_values(&["heartbeat", "success"]).inc();
+                let region_id = resp.get_region_id();
+                let epoch = resp.take_region_epoch();
+                let peer = resp.take_target_peer();
+
+                if resp.has_change_peer() {
+                    PD_HEARTBEAT_COUNTER_VEC.with_label_values(&["change peer"]).inc();
+
+                    let mut change_peer = resp.take_change_peer();
+                    info!("[region {}] try to change peer {:?} {:?}",
+                          region_id,
+                          change_peer.get_change_type(),
+                          change_peer.get_peer());
+                    let req = new_change_peer_request(change_peer.get_change_type().into(),
+                                                      change_peer.take_peer());
+                    send_admin_request_raw(&ch, region_id, epoch, peer, req);
+                } else if resp.has_transfer_leader() {
+                    PD_HEARTBEAT_COUNTER_VEC.with_label_values(&["transfer leader"]).inc();
+
+                    let mut transfer_leader = resp.take_transfer_leader();
+                    info!("[region {}] try to transfer leader from {:?} to {:?}",
+                          region_id,
+                          peer,
+                          transfer_leader.get_peer());
+                    let req = new_transfer_leader_request(transfer_leader.take_peer());
+                    send_admin_request_raw(&ch, region_id, epoch, peer, req)
+                }
+            })
+            .map_err(|e| panic!("unexpected error: {:?}", e));
+        handle.spawn(f);
+        self.is_hb_receiver_scheduled = true;
+    }
 }
 
 impl<T: PdClient> Runnable<Task> for Runner<T> {
     fn run(&mut self, task: Task, handle: &Handle) {
         debug!("executing task {}", task);
 
+        if !self.is_hb_receiver_scheduled {
```

Comment:
Later, in most cases, PD will not return the response if the region has nothing to change. Is it ok to check scheduler here?